Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1

```diff
@@ -0,0 +1,357 @@
+    Write-Host "`n"
+    Write-Host "*** To disable RDP access user must have atleast contributor access on cloud service(s) of Subscription: [$($SubscriptionId)] ***" -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host "`n" 
+    Write-Host "Validating whether the current user [$($currentSub.Account.Id)] has valid account type [User] to run the script for subscription [$($SubscriptionId)]..."
+    
+    # Safe Check: Checking whether the current account is of type User
+    if($currentSub.Account.Type -ne "User")
+    {
+        Write-Host "Warning: This script can only be run by user account type." -ForegroundColor $([Constants]::MessageType.Warning)
+        break;
+    }
+    
+    Write-Host "Successfully validated" -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host "`t"
+    Write-Host "Fetching cloud service(s)..."
+    $controlIds = "Azure_CloudService_SI_Disable_RemoteDesktop_Access"
+    
+    # Array to store resources
+    $resources = @()
+    $resourceType = "Microsoft.ClassicCompute/domainNames"
+
+    # If json path not given fetch all cloud service.
+    if([string]::IsNullOrWhiteSpace($Path))
+    {
+        $resources = Get-AzResource -ResourceType $resourceType
+    }
+    else
+    {
+        if (-not (Test-Path -Path $Path))
+        {
+            Write-Host "Error: Json file containing cloud service(s) detail not found for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            break;        
+        }
+
+        # Fetching cloud service details for remediation.
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.FailedControlSet
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId};
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($resourceDetails.ResourceDetails | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No cloud service(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+        $resourceDetails.ResourceDetails | ForEach-Object { 
+            try
+            {
+                $resources += Get-AzResource -ResourceId $_.ResourceId
+            }
+            catch
+            {
+                Write-Host "Valid resource group(s) or resource name(s) not found in input json file. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                break
+            }
+        }
+    }
+
+    $totalCloudService = ($resources | Measure-Object).Count
+    if($totalCloudService -eq 0)
+    {
+        Write-Host "No cloud service(s) found." -ForegroundColor $([Constants]::MessageType.Error);
+        Write-Host $([Constants]::DoubleDashLine)
+        break;
+    }
+
+    Write-Host "Total cloud service(s): [$($totalCloudService)]"
+    $folderPath = [Environment]::GetFolderPath("MyDocuments") 
```

Comment:
Do we change the path to `%temp% or the current directory?

Reply:
Done